openmls/src/framing/validation.rs

```diff
@@ -186,16 +186,17 @@ impl DecryptedMessage {
             // Preconfigured senders are not supported yet #106/#151.
             Sender::Preconfigured(_) => unimplemented!(),
             Sender::NewMember => {
-                // Since this allows only commits to have a sender type `Member`, it checks
-                // ValSem112
-                if let MlsContentBody::Commit(commit) = self.plaintext().content() {
-                    if let Some(path) = commit.path() {
-                        Ok(path.leaf_key_package().credential().clone())
-                    } else {
-                        Err(ValidationError::NoPath)
+                // Since this allows only commits or external Add proposals to have a sender type `NewMember`, it checks
+                // ValSem112 & ValSem113
+                match self.plaintext().content() {
+                    MlsContentBody::Commit(Commit { path, .. }) => path
+                        .as_ref()
+                        .map(|p| p.leaf_key_package().credential().clone())
+                        .ok_or(ValidationError::NoPath),
+                    MlsContentBody::Proposal(Proposal::Add(AddProposal { key_package })) => {
+                        Ok(key_package.credential().clone())
                     }
-                } else {
-                    Err(ValidationError::NotACommit)
+                    _ => Err(ValidationError::NotACommitOrExternalAddProposal),
```

Comment:
This looks like things are being mixed now. In fact, the sender types are redefined in #949 and I'm starting to think that issue should also be covered by this PR. That way we'd be up to spec (modulo the preconfigured senders).

Reply:
Will include #949 in this PR